ocaml/runtime/extern.c

```diff
@@ -1279,8 +1240,26 @@ intnat reachable_words_once(value root, enum reachable_words_traversal_mode mode
         } else {
           extern_update_location_with_data(h, new_mark);
         }
+
         /* The block contributes to the total size */
         size += 1 + sz;           /* header word included */
+        if (sizes_by_root_id) {
+          if (new_mark == Shared) {
+#pragma GCC diagnostic push
+            /* GCC complains that [mark] could be uninitialized. This can only happen
+             * if previously_marked returns false. But then new_mark is set to
+             * identifier which by assumption is != Shared. */
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
```

Comment:
I'd rather initialize `new_mark` and let the optimizer try and remove the dead store. Could even initialize it to a sentinel (like `RootUnprocessed`, which it will never legitimately be) and use an assertion to be sure it's set.

Reply:
Agreed, please remove the GCC pragmas. GCC is not the only compiler in use!

Reply:
I added an additional sentinel value `Invalid` - I think this has a clearer meaning that trying to reuse one of the other states, and there are plenty of unused values in reachable_words_node_state.

Reply:
> Agreed, please remove the GCC pragmas. GCC is not the only compiler in use!

Do we use any other compilers for this code, or is this a comment against compiler-specific pragmas in general?

Reply:
At Jane Street, we mostly use GCC and experiment a bit with Clang. Upstream, lots of compilers.